src/syntax/rules.rkt

```diff
@@ -15,9 +17,27 @@
 
 (define *rulesets* (make-parameter '()))
 
+(define (rule-ops-supported? rule)
+  (define (ops-in-expr expr)
+    (cond
+      [(list? expr) (if (member (car expr) (*loaded-ops*))
+                        (for/and ([subexpr (cdr expr)])
+                          (ops-in-expr subexpr))
+                        #f)]
+      [else #t]))
+  (and (ops-in-expr (rule-input rule)) (ops-in-expr (rule-output rule))))
+
+(define (prune-rules!)
+  (*rulesets* (for/list ([ruleset (*rulesets*)])
+                (cons (for/list ([rule (car ruleset)]
+                                 #:when (rule-ops-supported? rule))
+                        rule)
+                      (cdr ruleset)))))
+
 (define-syntax-rule (define-ruleset name groups [rname input output] ...)
-  (begin (define name (list (rule 'rname 'input 'output) ...))
-	 (*rulesets* (cons (cons name 'groups) (*rulesets*)))))
+  (begin
+    (define name (for/list ([r (list (rule 'rname 'input 'output) ...)]) r))
```

Comment:
What's the point of this change? It seems to me like the loop is just the identity.